### PR TITLE
Add domain-sockets support

### PR DESCRIPTION
**Pull request by tomasherman**
This is just a rough sketch of how it might work ... it seems it should be straight-forward to implement support for it. Let me know what you think :)

### CODE REVIEW

File: .bsp/sbt.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"name":"sbt","version":"1.4.9","bspVersion":"2.0.0-M5","languages":["scala"],"argv":["/Library/Java/JavaVirtualMachines/zulu-15.jdk/Contents/Home/bin/java","-Xms100m","-Xmx100m","-classpath","/usr/local/Cellar/sbt/1.4.7/libexec/bin/sbt-launch.jar","xsbt.boot.Boot","-bsp"]}
```
Comment (hamnis) [inline .bsp/sbt.json:1]: should not be commited, please add a .gitignore for .bsp

**Comment by hamnis** [comment]
Hello.

This looks like a nice addition.

I'm sure there could be some uses for this.

If you are willing to do the work, then I would mind adding this.

**Comment by tomasherman** [comment]
yeah it's definitely useful, you can get huge performance increase for a system where you talk to different service over localhost by using unix sockets (and i kinda ssume someone who is using netty-backend for http4s is after performance) ... ok let me see what i can come up with :)

**Comment by hamnis** [comment]
I didnt really mean to close this, that happened as part of renaming master to main.

**Pull request closed without merging**